### DISCUSSION TITLE
Future Non-Bluetooth Support?

### DISCUSSION THREAD

**Discussion by dur75945** [darthcloud/BlueRetro#99] [Ideas]
Just a thought really, but is it possible to add the capability for a wired connection with the esp in place of a Bluetooth connection using the micro usb port?

**Comment by ryanschlegl** [comment]
I've also been interested in this and looking at either porting to the ESP-32-S2 or S3 or integrating a USB host controller as per [here](https://www.youtube.com/watch?v=po3FBdY0GS4) . Downside with the S2 and S3 as I've found out is that you either miss out on Bluetooth on the S2 (one of the key features of BlueRetro) or at least classic Bluetooth on the S3. However I'm still looking into it as having an all in one multi-console adapter that I can use my USB fight sticks on as well as modern Bluetooth controllers would be a dream.

Keen to hear if anyone else is working on this or has ideas as BlueRetro is such an awesome project.

**Comment by darthcloud** [comment]
To answer OP the micro-usb is for the usb serial bridge, the ESP32 got no USB phy (and lets not talk about bit-bang)

@ryanschlegl I think the only way to go is with the S3. BlueRetro architecture require two cores.

The big question is will BLE become the norm for controllers? Xbox made the move already, will Nintendo and PlayStation follow?

Losing classic BT currently sucks.
